lib/utils/aws/aws.go

```diff
@@ -253,8 +255,41 @@ func FilterAWSRoles(arns []string, accountID string) (result []Role) {
 
 // Role describes an AWS IAM role for AWS console access.
 type Role struct {
+	// Name is the full role name with the entire path.
+	Name string `json:"name"`
 	// Display is the role display name.
 	Display string `json:"display"`
 	// ARN is the full role ARN.
 	ARN string `json:"arn"`
 }
+
+// Roles is a slice of roles.
+type Roles []Role
+
+// Sort sorts the roles by their display names.
+func (roles Roles) Sort() {
+	sort.Slice(roles, func(x, y int) bool {
+		return strings.ToLower(roles[x].Display) < strings.ToLower(roles[y].Display)
+	})
+}
+
+// FindRoleByARN finds the role with the provided ARN.
+func (roles Roles) FindRoleByARN(arn string) (Role, bool) {
+	for _, role := range roles {
+		if role.ARN == arn {
+			return role, true
+		}
+	}
+	return Role{}, false
+}
+
+// FindRolesByName finds all roles matching the provided name.
+func (roles Roles) FindRolesByName(name string) (result Roles) {
+	for _, role := range roles {
+		// Match either full name or the display name.
+		if role.Display == name || role.Name == name {
```

Comment:
The role.name  is never shown to a user during `tsh aws` flow (right now we are only printing `role.Display, role.ARN`) thus it will be hard for a user to provide the correct role name with full path.

Reply:
Yes. I was debating about this. The only reason I had it is for backward compatibility since the "full name" is used in previous `tsh` so in case someone scripting this, checking the "full name" here will not break. WDYT